src/main/java/com/google/jenkins/plugins/storage/client/StorageClient.java

```diff
@@ -0,0 +1,105 @@
+   * @param pattern Pattern to match object name to delete from bucket.
+   * @throws IOException If there was an issue calling the GCS API.
+   */
+  public void deleteFromBucket(String bucket, String pattern) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(bucket));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(pattern));
+    deleteFromBucketRequest(bucket, pattern).execute();
+  }
+
+  /**
+   * Helper method to return the delete request.
+   *
+   * @param bucket GCS bucket to delete from.
+   * @param pattern Pattern to match object name to delete from bucket.
+   * @return Delete request.
+   * @throws IOException If there was an issue calling the GCS API.
+   */
+  public Storage.Objects.Delete deleteFromBucketRequest(String bucket, String pattern)
+      throws IOException {
+    return storage.objects().delete(bucket, pattern);
+  }
+
+  /**
+   * Uploads item with path pattern to Google Cloud Storage bucket of name bucket.
+   *
+   * @param pattern Pattern to match object name to upload to bucket.
+   * @param bucket Name of the bucket to upload to.
+   * @param content InputStreamContent of desired file to upload.
+   * @throws IOException If there was an issue calling the GCS API.
+   */
+  public StorageObject uploadToBucket(String pattern, String bucket, InputStreamContent content)
+      throws IOException {
+    Preconditions.checkNotNull(content);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(bucket));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(pattern));
+    return uploadToBucketRequest(pattern, bucket, content).execute();
+  }
+
+  /**
+   * Helper method to return the insert request.
+   *
+   * @param pattern Pattern to match object name to upload to bucket.
+   * @param bucket Name of the bucket to upload to.
+   * @param content InputStreamContent of desired file to upload.
+   * @return Insert request.
+   * @throws IOException If there was an issue calling the GCS API.
+   */
+  public Storage.Objects.Insert uploadToBucketRequest(
+      String pattern, String bucket, InputStreamContent content) throws IOException {
+    return storage.objects().insert(bucket, null, content).setName(pattern);
+  }
+
+  /**
+   * Delete bucket from GCS with name bucket. Bucket must be empty.
+   *
+   * @param bucket Name of GCS bucket to delete.
+   * @throws IOException If there was an issue calling the GCS API.
+   */
+  public void deleteBucket(String bucket) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(bucket));
+    deleteBucketRequest(bucket).execute();
+  }
+
+  /**
+   * Helper method to return the delete request.
+   *
+   * @param bucket Name of GCS bucket to delete.
+   * @return Delete request.
+   * @throws IOException If there was an issue calling the GCS API.
+   */
+  public Storage.Buckets.Delete deleteBucketRequest(String bucket) throws IOException {
+    return storage.buckets().delete(bucket);
```

Comment:
Need precondition here.

Reply:
With these methods, I don't have preconditions since only another method calls them which has precondition checks. I have this request method to make the API call test-able.

Reply:
Currently yes. But it's prudent to think beyond the current use case and towards future uses. It's best to think defensively in these cases. Hence the need for the preconditions throughout.

Reply:
Done